Cannot compile  openocd v0.11.0 on ubuntu24.04

I have tried to compile openocd using this repository for testing on ubuntu24, but I cannot.

How to fix this problem ?

Error messages are shown below.

```bash
libtool: link: ( cd "src/jtag/.libs" && rm -f "libjtag.la" && ln -s "../libjtag.la" "libjtag.la" )
depbase=`echo src/flash/nor/numicro.lo | sed 's|[^/]*$|.deps/&|;s|\.lo$||'`;\
/bin/bash ./libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I.   -I./src -I./src -I./src/helper -DPKGDATADIR=\"/home/ubuntu24/.local/share/openocd\" -DBINDIR=\"/home/ubuntu24/.local/bin\" -I./jimtcl -I./jimtcl  -Wall -Wstrict-prototypes -Wformat-security -Wshadow -Wextra -Wno-unused-parameter -Wbad-function-cast -Wcast-align -Wredundant-decls -Wpointer-arith -Werror -g -O2 -MT src/flash/nor/numicro.lo -MD -MP -MF $depbase.Tpo -c -o src/flash/nor/numicro.lo src/flash/nor/numicro.c &&\
mv -f $depbase.Tpo $depbase.Plo
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I./src -I./src -I./src/helper -DPKGDATADIR=\"/home/ubuntu24/.local/share/openocd\" -DBINDIR=\"/home/ubuntu24/.local/bin\" -I./jimtcl -I./jimtcl -Wall -Wstrict-prototypes -Wformat-security -Wshadow -Wextra -Wno-unused-parameter -Wbad-function-cast -Wcast-align -Wredundant-decls -Wpointer-arith -Werror -g -O2 -MT src/flash/nor/numicro.lo -MD -MP -MF src/flash/nor/.deps/numicro.Tpo -c src/flash/nor/numicro.c -o src/flash/nor/numicro.o
src/flash/nor/numicro.c: In function 'numicro_fmc_cmd':
src/flash/nor/numicro.c:1244:17: error: this 'if' clause does not guard... [-Werror=misleading-indentation]
 1244 |                 if (retval != ERROR_OK)
      |                 ^~
In file included from ./src/flash/common.h:21,
                 from src/flash/nor/core.h:25,
                 from src/flash/nor/imp.h:24,
                 from src/flash/nor/numicro.c:32:
./src/helper/log.h:119:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'if'
  119 |         do { \
      |         ^~
src/flash/nor/numicro.c:1246:25: note: in expansion of macro 'LOG_DEBUG'
 1246 |                         LOG_DEBUG("status: 0x%" PRIx32 "", status);
      |                         ^~~~~~~~~
src/flash/nor/numicro.c: In function 'numicro_erase':
src/flash/nor/numicro.c:1513:25: error: this 'if' clause does not guard... [-Werror=misleading-indentation]
 1513 |                         if (retval != ERROR_OK)
      |                         ^~
./src/helper/log.h:119:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'if'
  119 |         do { \
      |         ^~
src/flash/nor/numicro.c:1515:33: note: in expansion of macro 'LOG_DEBUG'
 1515 |                                 LOG_DEBUG("status: 0x%" PRIx32 "", status);
      |                                 ^~~~~~~~~
src/flash/nor/numicro.c: In function 'numicro_write':
src/flash/nor/numicro.c:1602:33: error: this 'if' clause does not guard... [-Werror=misleading-indentation]
 1602 |                                 if (retval != ERROR_OK)
      |                                 ^~
./src/helper/log.h:119:9: note: ...this statement, but the latter is misleadingly indented as if it were guarded by the 'if'
  119 |         do { \
      |         ^~
src/flash/nor/numicro.c:1604:41: note: in expansion of macro 'LOG_DEBUG'
 1604 |                                         LOG_DEBUG("status: 0x%" PRIx32 "", status);
      |                                         ^~~~~~~~~
cc1: all warnings being treated as errors
make[3]: *** [Makefile:3615: src/flash/nor/numicro.lo] Error 1
make[3]: Leaving directory '/home/ubuntu24/.local/src/openocd'
make[2]: *** [Makefile:5297: install-recursive] Error 1
make[2]: Leaving directory '/home/ubuntu24/.local/src/openocd'
make[1]: *** [Makefile:5604: install] Error 2
make[1]: Leaving directory '/home/ubuntu24/.local/src/openocd'
make: *** [Makefile:57: openocd] Error 2
```

env:
- Ubuntu 24.04 LTS Desktop (x86_64, Linux 6.8.0-38-generic)
- additional installed pkgs `build-essential git gcc g++ autoconf automake libtool pkg-config make texinfo libusb-1.0-0-dev libhidapi-dev libgpiod-dev libftdi1-dev`

--- a/install/openocd/install.sh
+++ b/install/openocd/install.sh
@@ -14,7 +14,6 @@ _linux_install() {
   ./configure --prefix=$HOME/.local
   make
   make install
-  exec $SHELL -l #Reflects openocd installation.
 }
 
 archi=$(uname -sm)

--- a/install/openocd/install.sh
+++ b/install/openocd/install.sh
@@ -5,6 +5,8 @@ cd $(dirname $0)
 
 _linux_install() {
   # Install Open OCD
+  sudo apt update
+  sudo apt install -y make libtool pkg-config autoconf automake texinfo
   mkdir -p $HOME/.local/src && cd $HOME/.local/src
   git clone git://git.code.sf.net/p/openocd/code openocd -b v0.12.0
   cd openocd

--- a/install/openocd/install.sh
+++ b/install/openocd/install.sh
@@ -12,7 +12,9 @@ _linux_install() {
   cd openocd
   ./bootstrap
   ./configure --prefix=$HOME/.local
+  make
   make install
+  exec $SHELL -l #Reflects openocd installation.
 }
 
 archi=$(uname -sm)

--- a/install/openocd/install.sh
+++ b/install/openocd/install.sh
@@ -15,6 +15,8 @@ _linux_install() {
   make
   make install
   sudo cp contrib/60-openocd.rules /etc/udev/rules.d
+  sudo udevadm control --reload-rules
+  sudo udevadm trigger
 }
 
 archi=$(uname -sm)

--- a/install/openocd/install.sh
+++ b/install/openocd/install.sh
@@ -6,7 +6,7 @@ cd $(dirname $0)
 _linux_install() {
   # Install Open OCD
   mkdir -p $HOME/.local/src && cd $HOME/.local/src
-  git clone git://git.code.sf.net/p/openocd/code openocd -b v0.11.0
+  git clone git://git.code.sf.net/p/openocd/code openocd -b v0.12.0
   cd openocd
   ./bootstrap
   ./configure --prefix=$HOME/.local

--- a/install/openocd/install.sh
+++ b/install/openocd/install.sh
@@ -14,6 +14,7 @@ _linux_install() {
   ./configure --prefix=$HOME/.local
   make
   make install
+  sudo cp contrib/60-openocd.rules /etc/udev/rules.d
 }
 
 archi=$(uname -sm)

